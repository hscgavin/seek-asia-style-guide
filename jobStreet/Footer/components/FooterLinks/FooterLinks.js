import React from 'react';
import PropTypes from 'prop-types';

import styles from './FooterLinks.less';

const FooterLinks = ({ messages }) => {
  return (
    <ul className={styles.list}>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.link']}>
          { messages['footer.linkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.siteMapLink']}>
          { messages['footer.siteMapLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.aboutLink']}>
          { messages['footer.aboutLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.profilesLink']}>
          { messages['footer.profilesLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.termsLink']}>
          { messages['footer.termsLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.privacyLink']}>
          { messages['footer.privacyLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.safeLink']}>
          { messages['footer.safeLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.helpLink']}>
          { messages['footer.helpLinkText'] }
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.feedbackLink']}>
          { messages['footer.feedbackLinkText'] }
        </a>
      </li>
    </ul>
  );
};

FooterLinks.propTypes = {
  messages: PropTypes.object.isRequired
};

export default FooterLinks;
