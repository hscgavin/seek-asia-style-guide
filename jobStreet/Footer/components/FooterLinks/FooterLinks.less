.list {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  padding:45px 10px 35px;
  border-bottom: 1px solid #ccc;
}
.item {
  &:after {
    font-size: 1.1rem;
    content: '|';
    display: inline-block;
    color: #999;
  }
  &:last-child:after {
    display:none;
  }
}
.link {
  display: inline-block;
  padding:0 10px 10px;
  text-decoration: none;
  color: #1c3f94;
}