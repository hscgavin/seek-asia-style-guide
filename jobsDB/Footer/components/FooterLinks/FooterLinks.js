import React from 'react';
import PropTypes from 'prop-types';

import styles from './FooterLinks.less';

const FooterLinks = ({ messages }) => {
  return (
    <ul className={styles.list}>
      <li className={styles.itemMobile}>
        <a className={styles.link} href={messages['footer.legalLink']}>
          {messages['footer.legalLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.privacyLink']}>
          {messages['footer.privacyLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.termsLink']}>
          {messages['footer.termsLinkText']}
        </a>
      </li>
      <li className={styles.item}>
        <a className={styles.link} href={messages['footer.siteMapLink']}>
          {messages['footer.siteMapLinkText']}
        </a>
      </li>
    </ul>
  );
};

FooterLinks.propTypes = {
  messages: PropTypes.object.isRequired
};

export default FooterLinks;
