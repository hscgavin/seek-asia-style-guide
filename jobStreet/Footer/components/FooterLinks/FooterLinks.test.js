import React from 'react';
import { shallow } from 'enzyme';
import FooterLinks from './FooterLinks';

const baseUrl = 'https://jobstreet.com.my';
const baseUrlWithLanguage = `${baseUrl}/en`;

const messages = {
  'footer.copyright': 'Copyright Â© {year} JobStreet.com',
  'footer.link': 'https://www.jobstreet.com',
  'footer.linkText': 'JobStreet International',
  'footer.siteMapLink': `${baseUrl}/sitemap.htm`,
  'footer.siteMapLinkText': 'Site Map',
  'footer.aboutLink': `${baseUrlWithLanguage}/about-us/`,
  'footer.aboutLinkText': 'About Us',
  'footer.profilesLink': `${baseUrl}/career/JobStreet.htm`,
  'footer.profilesLinkText': 'Work With Us',
  'footer.termsLink': `${baseUrlWithLanguage}/about-us/terms-of-use/`,
  'footer.termsLinkText': 'Term of Use',
  'footer.privacyLink': `${baseUrlWithLanguage}/about-us/privacy-policy/`,
  'footer.privacyLinkText': 'Privacy Policy',
  'footer.safeLink': `${baseUrlWithLanguage}/about-us/safe-job-search-guide/`,
  'footer.safeLinkText': 'Safe Job Search Guide',
  'footer.helpLink': `${baseUrlWithLanguage}/user/`,
  'footer.helpLinkText': 'Help',
  'footer.feedbackLink':
    'https://myjobstreet.jobstreet.com.my/home/feedback.php?site=my&sub=feedback',
  'footer.feedbackLinkText': 'Send Feedback'
};

describe('FooterLinks component', () => {
  it('should render supplied messages', () => {
    const wrapper = shallow(<FooterLinks messages={messages} />);
    expect(wrapper).toMatchSnapshot();
  });
});
