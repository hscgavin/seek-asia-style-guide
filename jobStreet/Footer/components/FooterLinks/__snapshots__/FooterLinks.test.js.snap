// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinks component should render supplied messages 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://www.jobstreet.com"
    >
      JobStreet International
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/sitemap.htm"
    >
      Site Map
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/en/about-us/"
    >
      About Us
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/career/JobStreet.htm"
    >
      Work With Us
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/en/about-us/terms-of-use/"
    >
      Term of Use
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/en/about-us/privacy-policy/"
    >
      Privacy Policy
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/en/about-us/safe-job-search-guide/"
    >
      Safe Job Search Guide
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://jobstreet.com.my/en/user/"
    >
      Help
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      href="https://myjobstreet.jobstreet.com.my/home/feedback.php?site=my&sub=feedback"
    >
      Send Feedback
    </a>
  </li>
</ul>
`;
